### PR TITLE
Several refactoring patches as a preparation for the re-design of the orderbook in the context of tantalus + some cool features

### PR DESCRIPTION
**Pull request by thomaseizinger**
Review patch by patch is recommended.

### CODE REVIEW

File: cnd/src/network/comit_node.rs [hunk 1]
```diff
@@ -0,0 +1,692 @@
+use crate::{
```
Comment (D4nte) [inline cnd/src/network/comit_node.rs:1]: I assume this has just been moved from `network.rs` without functional changes :) (d631262d3a5c97cebdb910afb083c817cd7a7352)

File: cnd/src/network/swarm.rs [hunk 2]
```diff
@@ -0,0 +1,210 @@
+use crate::{
```
Comment (D4nte) [inline cnd/src/network/swarm.rs:1]: I assume this has just been moved from `network.rs` without functional changes :) (d631262d3a5c97cebdb910afb083c817cd7a7352)

**Review by D4nte — APPROVED** [review]

**Review by tcharding — APPROVED** [review]

File: cnd/src/network/comit_node.rs (hunk 1 above)
Comment (thomaseizinger) [reply cnd/src/network/comit_node.rs:1]: Yes, was moved without functional changes.

File: cnd/src/network/swarm.rs (hunk 2 above)
Comment (thomaseizinger) [reply cnd/src/network/swarm.rs:1]: Yes, was moved without functional changes.

**Comment by thomaseizinger** [comment]
@D4nte & @tcharding I've pushed another patch that hopefully makes e2e tests more stable.

**Comment by thomaseizinger** [comment]
@D4nte, @tcharding, @bonomat: This PR is no longer a pure refactoring.

I added an (IMO) cool feature where the user can now specify which COMIT "network" they want to connect to. For now, it doesn't to much other than influence the defaults of the configuration file but I believe this is in line with what we discussed on the mailing list at some point and it solves the problem I ran into with my earlier change.

More specifically, [`df19a33` (#3066)](https://github.com/comit-network/comit-rs/pull/3066/commits/df19a33d44b6b4c7eaa555d730251fba9b23cfa1) changes all defaults to "Mainnet". That however caused issues with our e2e test suite as some tests don't start up a blockchain environment (sanity tests). However, when run all together, the `cnd` instances started by these tests still see the environment from the other tests and connect to it on the default ports. Due to the new config defaults, the network validation fails and cnd exits :D

With [`1113be7` (#3066)](https://github.com/comit-network/comit-rs/pull/3066/commits/1113be713b63b8030effde926315e8cdc68c1582), we can specify `--network dev` on the command line and hence default the config to `dev` even for those tests!

File: cnd/src/config.rs [hunk 3]
```diff
@@ -4,31 +4,56 @@ mod settings;
...
     file::File,
     settings::{AllowedOrigins, Settings},
     validation::validate_connection_to_network,
 };
+use crate::data_dir;
 
-lazy_static::lazy_static! {
-    pub static ref LND_URL: Url = Url::parse("https://localhost:8080").expect("static string to be a valid url");
-}
+static BITCOIND_RPC_MAINNET: Lazy<Url> = Lazy::new(|| parse_unchecked("http://localhost:8332"));
```
Comment (D4nte) [inline cnd/src/config.rs:21]: commit message of e908b28b5cc12bc65e30a6e68e506d7f3689d437 does not match the patch content.

File: cnd/src/config.rs [hunk 4]
```diff
@@ -180,11 +221,11 @@ fn assert_lnd_url_https(lnd_url: Url) -> Result<Url> {
     }
 }
 
-impl Default for Lightning {
-    fn default() -> Self {
+impl Lightning {
```
Comment (D4nte) [inline cnd/src/config.rs:224]: I am not sure having two `impl` blocks really make sense.

File: comit/src/lib.rs [hunk 5]
```diff
@@ -119,12 +119,47 @@ pub enum Role {
...
+#[derive(Clone, Copy, Debug, strum_macros::Display, strum_macros::EnumString, PartialEq, Eq)]
 pub enum Side {
     Alpha,
     Beta,
 }
 
+/// The various networks within COMIT.
+#[derive(Debug, Clone, Copy, strum_macros::Display, strum_macros::EnumString, PartialEq, Eq)]
+#[strum(serialize_all = "lowercase")]
+pub enum Network {
```
Comment (D4nte) [inline comit/src/lib.rs:131]: "comit network" finally!

**Review by D4nte — APPROVED** [review]

File: cnd/src/config.rs (hunk 4 above)
Comment (thomaseizinger) [reply cnd/src/config.rs:224]: It doesn't! :D

Left over from amending a lot :grimacing:

File: cnd/src/config.rs (hunk 3 above)
Comment (thomaseizinger) [reply cnd/src/config.rs:21]: Fair, I've probably squashed too much into this one :)

**Review by bonomat — APPROVED** [review]
I can't say much about the introduced changes :) besides good job on introducing COMIT network. 
We might want to follow-up on the discussion around the concept of the COMIT network as I can see some limitations (e.g. how can we introduce other networks such as Bitcoin's Signet?).

**Comment by thomaseizinger** [comment]
> We might want to follow-up on the discussion around the concept of the COMIT network as I can see some limitations (e.g. how can we introduce other networks such as Bitcoin's Signet?).

Yes, there is definitely potential for improving on this. The same applies for choosing between the plethora of Ethereum chains that can be used for testing :)

Happy to discuss this on the mailing list!

**Pull request merged**